### PR TITLE
[Merged by Bors] - chore(Order.Filter): remove duplicated lemmas

### PR DESCRIPTION
**Pull request by sgouezel**
Remove the lemmas `le_pure_iff_eq_pure` (duplicate of `NeBot.le_pure_iff`) and `eq_bot_or_pure_of_subsingleton_mem` (duplicate of `subsingleton_iff_bot_or_pure`). 

This requires moving a few lemmas after their non-duplicate prerequisites.

---

As pointed out on Zulip (https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.236864/near/388779678) by @urkud.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Filter/Subsingleton.lean [hunk 1]
```diff
@@ -72,3 +72,8 @@ theorem subsingleton_iff_exists_singleton_mem [Nonempty α] : l.Subsingleton ↔
 
 /-- A subsingleton filter on a nonempty type is less than or equal to `pure a` for some `a`. -/
 alias ⟨Subsingleton.exists_le_pure, _⟩ := subsingleton_iff_exists_le_pure
+
+lemma isCountablyGenerated_of_subsingleton (hl : l.Subsingleton) : IsCountablyGenerated l := by
```
Comment (eric-wieser) [inline Mathlib/Order/Filter/Subsingleton.lean:76]: Should this be
```suggestion
lemma Subsingleton.isCountablyGenerated (hl : l.Subsingleton) : IsCountablyGenerated l := by
```
for dot notation?

File: Mathlib/Order/Filter/Ultrafilter.lean [hunk 2]
```diff
@@ -414,6 +414,22 @@ protected theorem NeBot.le_pure_iff (hf : f.NeBot) : f ≤ pure a ↔ f = pure a
...
 #align filter.ne_bot.le_pure_iff Filter.NeBot.le_pure_iff
 
+protected theorem NeBot.eq_pure_iff (hf : f.NeBot) {x : α} :
+    f = pure x ↔ {x} ∈ f := by
+  rw [← hf.le_pure_iff, le_pure_iff]
+
+lemma atTop_eq_pure_of_isTop [LinearOrder α] {x : α} (hx : IsTop x) :
+    (atTop : Filter α) = pure x := by
+  have : Nonempty α := ⟨x⟩
+  have A : (atTop : Filter α).NeBot := atTop_basis.neBot_iff.2 (fun _ ↦ ⟨x, hx _⟩)
```
Comment (ADedecker) [inline Mathlib/Order/Filter/Ultrafilter.lean:424]: Can't you use [Filter.atTop_neBot](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/AtTopBot.html#Filter.atTop_neBot) here?

Comment (sgouezel) [reply Mathlib/Order/Filter/Ultrafilter.lean]: Better, thanks! I think it was just the output suggested by `exact?`, and I didn't try to simplify it further.

**Comment by ADedecker** [comment]
Thanks!
bors merge

**Pull request closed without merging**